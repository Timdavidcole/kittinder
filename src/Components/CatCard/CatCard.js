import React from 'react'
import agent from '../../agent'

class CatCard extends React.Component {
  constructor (props) {
    super(props)
    this.state = {
      image: '',
      resolved: false,
      loading: false,
      error: null
    }
  }

  componentDidMount () {
    agent.Images.get({ breed_id: this.props.cat.id }).then(data =>
      this.setState({ image: data[0].url })
    )
  }

  render () {
    const {cat} = this.props
    return (
      <div>
        <h2>{cat.name}</h2>
        <img
          src={this.state.image}
          alt={''}
          style={{ maxWidth: '100%', maxHeight: '100vh' }}
        ></img>
        <div style={{ width: '100%' }}>
          <span>{cat.description}</span>
        </div>
      </div>
    )
  }

}

export default CatCard
