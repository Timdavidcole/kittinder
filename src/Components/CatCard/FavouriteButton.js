import React from "react";
import agent from "../../agent";

class FavouriteButton extends React.Component {
  constructor(props) {
    super(props);
    this.state = { error: false }
    this.onClick = this.onClick.bind(this);
  }

  onClick() {
    agent.Favourites.post(this.props.imageId, this.props.userId)
      .then(data => null)
      .catch(error => {
        this.setState({
          error: true
        });
      });
  }

  render() {
    return (
      <button
        id={`favouriteButton${this.props.imageId}`}
        onClick={this.onClick}
      >
        Favourite
      </button>
    );
  }
}

export default FavouriteButton;
