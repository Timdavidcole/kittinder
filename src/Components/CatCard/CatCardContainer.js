import React from "react";
import shuffle from "../../Models/shuffle";
import CatCard from "./CatCard";


class CatCardContainer extends React.Component {
  render() {
    return (
      <div className="catCardContainer" style={{ padding: "10px" }}>
        {shuffle(this.props.cats).map((cat, index) => {
          return <CatCard cat={cat} key={index} />;
        })}
      </div>
    );
  }
}

export default CatCardContainer;
