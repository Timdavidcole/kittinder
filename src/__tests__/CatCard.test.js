import CatCard from "../Components/CatCard/CatCard";
import React from "react";
import renderer from "react-test-renderer";
import "@testing-library/jest-dom";
import { render, screen } from "@testing-library/react";
import { mount, configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

configure({ adapter: new Adapter() });

jest.mock("../agent");
const testCat = { name: "test cat", description: "I'm a test cat" };
test("checks CatCard snapshot", () => {
  const catCard = renderer.create(<CatCard cat={testCat} />);

  let tree = catCard.toJSON();
  expect(tree).toMatchSnapshot();
});

test("CatCard has correct text", () => {
  render(<CatCard cat={testCat} />);

  expect(screen.queryByText(testCat.description)).toBeInTheDocument();
  expect(screen.queryByText(testCat.name)).toBeInTheDocument();
});

test("CatCard has correct image url", done => {
  const wrapper = mount(<CatCard cat={testCat} />);
  setImmediate(() => {
    expect(wrapper.state().image).toBe("test");
    done();
  });
});
