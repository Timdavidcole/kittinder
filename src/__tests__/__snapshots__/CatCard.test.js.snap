// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CatCard renders 1`] = `
<div>
  <h2>
    test cat
  </h2>
  <img
    alt=""
    id="cat-image"
    src=""
    style={
      Object {
        "maxHeight": "100vh",
        "maxWidth": "100%",
      }
    }
  />
  <div
    style={
      Object {
        "width": "100%",
      }
    }
  >
    <span>
      I'm a test cat
    </span>
  </div>
</div>
`;

exports[`checks CatCard snapshot 1`] = `
<div>
  <h2>
    test cat
  </h2>
  <img
    alt=""
    id="cat-image"
    src=""
    style={
      Object {
        "maxHeight": "100vh",
        "maxWidth": "100%",
      }
    }
  />
  <div
    style={
      Object {
        "width": "100%",
      }
    }
  >
    <span>
      I'm a test cat
    </span>
  </div>
</div>
`;
